### ISSUE TITLE
Why batch size is 1 ?

### ISSUE THREAD

**Issue by JerryLauzzz** [junyanz/CycleGAN#55]
When using batch norm, batch size more than 16 is better I think, to reduce variance.
Why the batch size is 1 here ?

**Comment by phillipi** [comment]
What we are doing here is instance norm (i.e. batchnorm with batch size 1 and no running mean/variance: https://arxiv.org/abs/1607.08022). Instance norm has the effect of making the output invariant to mean and variance of each feature channel of the input. This is the same idea as contrast normalization. Note that this only makes sense if each feature channel has spatial size greater than 1x1, since otherwise subtracting the mean will zero the features.

**Comment by JerryLauzzz** [comment]
Thanks.

**Comment by fanpengustc** [comment]
Instance has nothing to do with batch size, I also confused about why batch size is 1

**Comment by phillipi** [comment]
Can you elaborate? In the context of convolutional features, batch norm with batch size 1 applied at both train and test time is, I believe, equivalent to instance norm. Please see this paper for more details: https://arxiv.org/pdf/1607.08022.pdf

One other reason we use batch size 1 is that allows us to fit bigger images in memory.

**Comment by swang423** [comment]
Using such a small batch size slows down training significantly if there are many images. Have you tried larger batch sizes? do you need to change Instance Normalization to batch normalization if you increase the batch size to 32 or even 256?

**Comment by junyanz** [comment]
You can use instance norm.
